ethereum/transaction_status_task.go

```diff
@@ -167,13 +224,21 @@ func (tst *TransactionStatusTask) RunTask() (resp interface{}, err error) {
 	}
 
 	err = tst.isTransactionSuccessful(ctx, tst.txHash)
-	if err == nil {
-		return nil, nil
+	if err != nil {
+		if err != ErrTransactionFailed {
+			err = gocelery.ErrTaskRetryable
+		}
+		return nil, err
 	}
 
-	if err != ErrTransactionFailed {
-		return nil, gocelery.ErrTaskRetryable
+	if tst.eventName != "" {
+		v, err := tst.getEventValueFromTransactionReceipt(ctx, tst.txHash, tst.eventName, tst.eventValueIdx)
```

Comment:
okay, so we always add the events to the centrifuge transactions.

Reply:
Not always, only when there an event filter passed to the task.